core/spec/models/spree/order_cancellations_spec.rb

```diff
@@ -152,7 +152,9 @@
 
       it "generates the correct total amount" do
         order.cancellations.short_ship([inventory_unit_1])
+        inventory_unit_2.line_item.reload # UnitCancel#compute_amount needs updated amounts
         order.cancellations.short_ship([inventory_unit_2])
+        line_item.reload
```

Comment:
Here's an idea for getting OrderUpdater to operate on in-memory objects, to try to avoid problems like the above.
```diff
diff --git a/core/app/models/spree/order_updater.rb b/core/app/models/spree/order_updater.rb
index caff2a6..b21a70c 100644
--- a/core/app/models/spree/order_updater.rb
+++ b/core/app/models/spree/order_updater.rb
@@ -31,7 +31,15 @@ module Spree
     end
 
     def recalculate_adjustments
-      all_adjustments.includes(:adjustable).map(&:adjustable).uniq.each { |adjustable| Spree::ItemAdjustments.new(adjustable).update }
+      adjustables = Set.new
+
+      # Prefer using in-memory objects when possible:
+      adjustables << order
+      adjustables.merge(order.line_items)
+      adjustables.merge(order.shipments)
+      # In case someone is attaching adjustments to something other than LineItems and Shipments:
+      adjustables.merge(all_adjustments.includes(:adjustable).map(&:adjustable))
+
+      adjustables.each do |adjustable|
+        Spree::ItemAdjustments.new(adjustable).update
+      end
     end
```
Two risks I can think of:
1) The in-memory objects may be stale.  That might be an acceptable risk in the interest of performance.
2) The above code will invoke `ItemAdjustments#update` on all shipments & line items, even if they don't currently have any adjustments on them. That will result in an extra `select * from spree_adjustments where adjustable_id = ?` for every line item/shipment that doesn't have at least one adjustment (extra compared to the existing code).

Any thoughts?